rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java

```diff
@@ -816,6 +816,45 @@ Serializable valueOfTaskResult(@HeaderParam("sessionid") String sessionId, @Path
 			@PathParam("jobid") String jobId, @PathParam("tasktag") String taskTag) throws Throwable;
 
 	/**
+	 * Returns the metadata of the task result of task <code>taskName</code> of the
+	 * job <code>jobId</code>.
+	 *
+	 * Metadata is a map containing additional information associated with a result. For example a file name if the result represents a file.
+	 *
+	 * @param sessionId
+	 *            a valid session id
+	 * @param jobId
+	 *            the id of the job
+	 * @param taskname
+	 *            the name of the task
+	 * @return the metadata of the task result
+	 */
+	@GET
+	@GZIP
+	@Path("jobs/{jobid}/tasks/{taskname}/result/metadata")
+	@Produces("*/*")
+	Map<String, String> metadataOfTaskResult(@HeaderParam("sessionid") String sessionId, @PathParam("jobid") String jobId,
+											 @PathParam("taskname") String taskname) throws Throwable;
+
+	/**
+	 * Returns the metadata of the task result of task <code>taskName</code> of the
+	 * job <code>jobId</code>filtered by a given tag.
+	 * <p>
+	 * Metadata is a map containing additional information associated with a result. For example a file name if the result represents a file.
+	 *
+	 * @param sessionId a valid session id
+	 * @param jobId     the id of the job
+	 * @param taskTag   the tag used to filter the tasks.
+	 * @return a map containing for each task entry, the metadata of the task result
+	 */
+	@GET
+	@GZIP
+	@Path("jobs/{jobid}/tasks/tag/{tasktag}/result/metadata")
+	@Produces("application/json")
+	Map<String, Map<String, String>> metadataOfTaskResultByTag(@HeaderParam("sessionid") String sessionId,
+															   @PathParam("jobid") String jobId, @PathParam("tasktag") String taskTag) throws Throwable;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00112)